### PR TITLE
Remove letter code from tasks.py , related tests and imports

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

Cleaned up letter code.
Removed code from app/tasks.py, test_tasks.py and all unused imports.

Removed all periphery code that called functions from app/tasks.py

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,7 @@ format:
 	isort .
 	black --config pyproject.toml .
 	flake8 .
+	mypy .
```
Comment (andrewleith) [inline Makefile:56]: Do we need to run `mypy` here?  If so, should we do the same in admin?  `mypy` is currently run as part of `make test` (in `run_scripts.sh`).

File: app/celery/tasks.py [hunk 2]
```diff
@@ -34,24 +33,14 @@
     put_batch_saving_bulk_created,
     put_batch_saving_bulk_processed,
 )
-from app.celery import (  # noqa: F401
```
Comment (andrewleith) [inline app/celery/tasks.py:37]: Any idea why these were imported if not used?

File: app/letters/rest.py [hunk 3]
```diff
@@ -14,6 +12,4 @@
 def create_process_returned_letters_job():
```
Comment (andrewleith) [inline app/letters/rest.py:12]: Should this whole method be removed?

File: app/notifications/notifications_letter_callback.py [hunk 4]
```diff
@@ -55,7 +50,5 @@ def process_letter_response():
 
         if filename.lower().endswith("rs.txt") or filename.lower().endswith("rsp.txt"):
             current_app.logger.info("DVLA callback: Calling task to update letter notifications")
-            update_letter_notifications_statuses.apply_async([filename], queue=QueueNames.NOTIFY)
```
Comment (andrewleith) [inline app/notifications/notifications_letter_callback.py:58]: Should this whole method be removed?

File: tests/app/celery/test_ftp_update_tasks.py [hunk 5]
```diff
@@ -44,182 +16,6 @@ def notification_update():
...
-            update_letter_notifications_to_error([first.reference])
-    assert first.reference in str(e.value)
-
-    assert first.status == NOTIFICATION_TECHNICAL_FAILURE
-    assert first.sent_by is None
-    assert first.sent_at is None
-    assert first.updated_at == dt
-    assert second.status == NOTIFICATION_CREATED
-
-
```
Comment (andrewleith) [inline tests/app/celery/test_ftp_update_tasks.py:222]: So long, letter tests!

File: tests/app/letters/test_returned_letters.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import pytest
 
 
+@pytest.mark.skip(reason="Deprecated: LETTER CODE")
```
Comment (andrewleith) [inline tests/app/letters/test_returned_letters.py:4]: Just curious why you are marking some things for deprecation and removing others?  Just the process you are following?

**Review by andrewleith — COMMENTED** [review]
Overall looks good, just a few questions.

File: tests/app/letters/test_returned_letters.py (hunk 6 above)
Comment (jzbahrai) [reply tests/app/letters/test_returned_letters.py:4]: this is good question! I basically was only concentrating on app/tasks.py and test_tasks.py. These two were the instigator. Now if the code from them was called else where I was deleting that code as well. In the above case - the actual rest API hasn't been removed but as it was mocking part of the tasks celery code, we were causing a failure. When the rest API is removed, I was thinking we can remove these further tests.

File: app/notifications/notifications_letter_callback.py (hunk 4 above)
Comment (jzbahrai) [reply app/notifications/notifications_letter_callback.py:58]: yes it should, but then the tests needs to be removed. So falling the principle I was using below - I didn't remove it.

File: app/letters/rest.py (hunk 3 above)
Comment (jzbahrai) [reply app/letters/rest.py:12]: yes it should, but then the tests needs to be removed. So following the principle I was using below - I didn't remove it.

File: app/celery/tasks.py (hunk 2 above)
Comment (jzbahrai) [reply app/celery/tasks.py:37]: unsure :/

File: Makefile (hunk 1 above)
Comment (jzbahrai) [reply Makefile:56]: I would add it in admin as well. its part of scripts/run_tests but wasn't part of `make format` . I generally run `make format` before I push

File: tests/app/letters/test_returned_letters.py (hunk 6 above)
Comment (andrewleith) [reply tests/app/letters/test_returned_letters.py:4]: Makes sense, thanks for the clarification.

**Review by andrewleith — APPROVED** [review]
LGTM

**Pull request merged**